rust_src/src/ng_async.rs

```diff
@@ -63,11 +66,59 @@ fn nullptr() -> usize {
     std::ptr::null() as *const i32 as usize
 }
 
+fn object_to_option(object: LispObject) -> PipeOptions {
+    match object {
+	Qstring => String::option(),
+	Quser_ptr => UserData::option(),
+	_ => panic!("None supported type for data transfer to async worker."),
+    }
+}
+
+fn option_to_object(option: PipeOptions) -> LispObject {
+    match option {
+	PipeOptions::STRING => Qstring,
+	PipeOptions::USER_DATA => Quser_ptr,
+    }
+}
```

Comment:
Maybe these functions should be methods ?

Reply:
Agreed. 

In addition I want to re-work my naming after thinking more about it. Having a PipeOption struct and a PipeOptions trait (which PipeOption doesn't implement) is very confusing.